File: fastcampus/01.완전탐색/15652.js
let fs = require('fs');
// let input = fs.readFileSync('/dev/stdin').toString();
let input = fs.readFileSync('ex.txt').toString();

input = input.split(' ')

let [n, m] = input.map(el => parseInt(el))
let arr = [];
let result = [];

function rec(k){
  if(arr.length === m){
    result.push(arr.join(' '))
    return
  }
  for(let i=k; i <= n; i ++){
    arr.push(i)
    rec(i)
    arr.pop()
  }
}

rec(1)
console.log(result.join('\n'))

File: fastcampus/01.완전탐색/15652.py
import sys
sys.setrecursionlimit(10**8)
input = sys.stdin.readline

n, m = map(int, input().split())

def rec(results):
    if len(results) == m:
        print(' '.join(map(str, results)))
        return
    if len(results) == 0:
        for i in range(1, n+1):
            results.append(i)
            rec(results)
            results.pop()
    else:
        for i in range(results[-1], n+1):
            results.append(i)
            rec(results)
            results.pop()

rec([])


